scc/provider/resource_subaccount_using_auth.go

```diff
@@ -321,11 +333,23 @@ func (r *SubaccountUsingAuthResource) Update(ctx context.Context, req resource.U
 		return
 	}
 
-	// if shouldUpdateTunnelCopy(plan) {
-	// 	if err := r.updateTunnelState(ctx, plan, state, endpoint, &respObj, &resp.Diagnostics); err != nil {
-	// 		return
-	// 	}
-	// }
+	if !plan.Connected.IsNull() && !plan.Connected.IsUnknown() {
+		desiredState := plan.Connected.ValueBool()
+		if desiredState != state.Connected.ValueBool() {
+			patch := map[string]any{"connected": desiredState}
+
+			if err := requestAndUnmarshal(r.client, &respObj, "PUT", endpoint+"/state", patch, false); err != nil {
+				resp.Diagnostics.AddError(errMsgUpdateSubaccountFailed, err.Error())
+				return
+			}
+
+			// Re-fetch to update tunnel state
+			if err := requestAndUnmarshal(r.client, &respObj, "GET", endpoint, nil, true); err != nil {
+				resp.Diagnostics.AddError(errMsgUpdateSubaccountFailed, err.Error())
+				return
+			}
+		}
+	}
 
 	if respObj.Tunnel.State == "Connected" {
```

Comment:
Shouldn't we issue a warning if the tunnel state has the value `connection_failure`?